### DISCUSSION TITLE
PostgreSQL: Execute INSERT with float params not working

### DISCUSSION THREAD

**Discussion by jgauchel** [sqlalchemy/sqlalchemy#6026] [Usage Questions]
**Describe the bug**
Executing a raw sql command with session.execute and float param values on a postgresql table with column type "double precision" does not work.

**To Reproduce**
table **user_numbers**: 
| field_name | type | nullable |
| - | - | - |
| id | integer | false |
| user_id | integer | false |
| value1 | double precision | true |
| value2 | double precision | true |
| value3 | double precision | true |

```python
session.execute(
    'INSERT INTO user_numbers (id, user_id, value1, value2, value3) VALUES(:id, :user_id, :value1, :value2, :value3'
    {'id': 1, 'user_id': 1, 'value1': 90, 'value2': 89.9216272997926, 'value3': 90.096095280000014}
)
```

**Expected behavior**
Table user_numbers should have a row with values inserted before. But the table stays empty.

If the float param values gets injected directly into sql string, it works as expected:
```python
value1 = 90
value2 = 89.9216272997926
value3 = 90.096095280000014

session.execute(
    f'INSERT INTO user_numbers (id, user_id, value1, value2, value3) VALUES(:id, :user_id, {value1}, {value2}, {value3}'
    {'id': 1, 'user_id': 1}
)
```

**Versions.**
 - OS: WSL 2
 - Python: 3.8.5
 - SQLAlchemy: 1.3.23
 - Database: PostgreSQL 12

**Have a smooooth day!**

**Comment by CaselIT** [comment]
Hi.
Sorry to be blunt, but could you provide a stack trace of the error and a reproduction case? 

thanks

**Comment by jgauchel** [comment]
There is no error (which caused some trouble...)! The table just stays empty after executing the insert sql command. You can check this by running 
```py
session.execute('SELECT * FROM user_numbers')
```
What else are you missing to reproduce the case?

**Comment by CaselIT** [comment]
If there is no error and the table is empty you probably need to call `session.commit()` to commit the transaction.

**Reply by jgauchel** [reply]
In my experience actually not.... and I tried it with and without `session.commit()`. Anyway.. I will provide a full python script, that needs postgresql connection only, to reproduce that case

**Reply by CaselIT** [reply]
Unless you set the session with `autocommit=True`, each statement is run in a transaction, so it needs to be manually committed.

> I will provide a full python script, that needs postgresql connection only, to reproduce that case

that would help a lot to understand what's going on, since I could not reproduce this case when I tried

**Reply by jgauchel** [reply]
ah! thanks for the info

**Reply by CaselIT** [reply]
you can pass `echo=True` to the `create_engine` to look at which statements are run by sqlalchemy.
